PancakeSwap-Bots/Large-swap/src/agent.ts

```diff
@@ -0,0 +1,93 @@
+import { Finding, HandleTransaction, TransactionEvent, ethers, getEthersProvider } from "forta-agent";
+
+import {
+  SWAP_EVENT,
+  isValidPancakePair,
+  getERC20Balance,
+  LARGE_THRESHOLD,
+  createFinding,
+  toBn,
+  PANCAKE_FACTORY_ADDRESS,
+} from "./utils";
+import BigNumber from "bignumber.js";
+BigNumber.set({ DECIMAL_PLACES: 18 });
+
+export const provideBotHandler = (
+  largePercentage: string,
+  pancakeFactory: string,
+  provider: ethers.providers.Provider
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    // filter the transaction logs for swap events
+    const swapEvents = txEvent.filterLog(SWAP_EVENT);
+    try {
```

Comment:
the `try/catch` should be inside the map, not here. Even better will be to have it in the `isValidPancakePair` function so if something fails it returns `[false, "", ""]`